sunpy/coordinates/spice.py

```diff
@@ -0,0 +1,169 @@
+
+
+# Defined for future use
+class _SpiceBaseCoordinateFrame(SunPyBaseCoordinateFrame):
+    pass
+
+
+def _convert_to_et(time):
+    return (time - _ET_REF_EPOCH).to_value('s')
+
+
+def _install_frame(frame_id):
+    frame_name = spiceypy.frmnam(frame_id)
+    # TODO: Sanitize/escape the frame name of special characters
+
+    frame_center, class_num, _ = spiceypy.frinfo(frame_id)
+    log.info(f"Installing {frame_name} {_CLASS_TYPES[class_num]} frame ({frame_id}) "
+             f"as 'spice_{frame_name}'")
+
+    spice_frame = type(f"spice_{frame_name}",
+                       (_SpiceBaseCoordinateFrame,),
+                       {})
+    # Force the capitalization
+    # TODO: Consider adding the alias of all lowercase
+    spice_frame.name = spice_frame.__name__
+
+    # TODO: Figure out how to handle array time
+    # TODO: Does it matter what time is used for J2000 ET?
+
+    @frame_transform_graph.transform(FunctionTransformWithFiniteDifference, ICRS, spice_frame)
+    def icrs_to_spice(from_icrs_coord, to_spice_frame):
+        et = _convert_to_et(to_spice_frame.obstime)
+        matrix = spiceypy.pxfrm2('J2000', frame_name, 0, et)
+        icrs_offset = spiceypy.spkezp(frame_center, et, 'J2000', 'none', 0)[0] << u.km
+        shifted_old_pos = from_icrs_coord.cartesian - CartesianRepresentation(icrs_offset)
+        new_pos = shifted_old_pos.transform(matrix)
+        return to_spice_frame.realize_frame(new_pos)
+
+    @frame_transform_graph.transform(FunctionTransformWithFiniteDifference, spice_frame, ICRS)
+    def spice_to_icrs(from_spice_coord, to_icrs_frame):
+        et = _convert_to_et(from_spice_coord.obstime)
+        matrix = spiceypy.pxfrm2(frame_name, 'J2000', et, 0)
+        shifted_new_pos = from_spice_coord.cartesian.transform(matrix)
+        icrs_offset = spiceypy.spkezp(frame_center, et, 'J2000', 'none', 0)[0] << u.km
+        new_pos = shifted_new_pos + CartesianRepresentation(icrs_offset)
+        return to_icrs_frame.realize_frame(new_pos)
+
+    frame_transform_graph._add_merged_transform(spice_frame, ICRS, spice_frame)
+
+
+# TODO: Support multiple calls to initialize()
+
+def initialize(kernels):
+    """
+    Load one more more SPICE kernels and create corresponding frame classes.
+
+    .. warning::
+        As currently implemented, do not run this more than once per Python
+        session.
+
+    Parameters
+    ----------
+    kernels : `str`, `list` of `str`
+         One or more SPICE kernel files
+
+    Notes
+    -----
+    If a kernel file is a meta-kernel, make sure that the relative paths therein
+    are correct for the current working directory, which may not be the same as the
+    location of the meta-kernel file.
+    """
+    spiceypy.furnsh(kernels)
+
+    for class_num in range(1, 7):
+        frames = spiceypy.kplfrm(class_num)
+        for frame_id in frames:
+            _install_frame(frame_id)
+
+
+# TODO: Add support for array time
+# TODO: Add support for light travel time correction
```

Comment:
If these TODOs don't end up in this first PR, let's make sure to convert these to issues so we don't lose track of them!

Reply:
Heh, I addressed all of my TODOs.  I'm done now, right? =)